packages/report-server/src/auth/middleware.ts

```diff
@@ -40,14 +40,7 @@ const authenticateBearerAuthHeader = async (authHeader: string) => {
 };
 
 const authenticateBasicAuthHeader = async (req: ReportsRequest, authHeader: string) => {
-  const usernamePassword = Buffer.from(authHeader.split(' ')[1], 'base64').toString();
-  if (!usernamePassword.includes(':')) {
-    throw new UnauthenticatedError('Invalid Basic auth credentials');
-  }
-
-  // Split on first occurrence because password can contain ':'
-  const username = usernamePassword.split(':')[0];
-  const password = usernamePassword.substring(username.length + 1, usernamePassword.length);
+  const { username, password } = getUserAndPassFromBasicAuth(authHeader);
```

Comment:
The other option would be to use `basic-auth` package like we do in `meditrak-server` (see `clientAuth.js`)

Reply:
But I'm kinda into rolling our own for something so straightforward, so actually how about instead updating `meditrak-server` to use yours?